File: python/src/easy/349/solution.py
from src.helper import Fields, tester

test_data = [
    {Fields.args: ([1,2,2,1],[2,2]), Fields.expd: [2]},
    {Fields.args: ([4,9,5],[9,4,9,8,4]), Fields.expd: [9,4]},
]


class Solution:
    def intersection(self, nums1: list[int], nums2: list[int]) -> list[int]:  # noqa: N802
        return list(set(nums1).intersection(nums2))


if __name__ == "__main__":
    solution = Solution()
    tester(func=solution.intersection, test_data=test_data)

File: go/easy/349/solution.go
package main

func intersection(nums1 []int, nums2 []int) []int {
	set1 := make(map[int]bool)
	for _, num := range nums1 {
		set1[num] = true
	}

	set2 := make(map[int]bool)
	for _, num := range nums2 {
		set2[num] = true
	}

	result := make([]int, 0, len(set1))
	for num := range set1 {
		if set2[num] {
			result = append(result, num)
		}
	}

	return result
}
